### ISSUE TITLE
[Icon] Mastodon for domains other than social.lol (and as a manual tag)

### ISSUE THREAD

**Issue by rosswintle** [neatnik/omg.lol#360]
### What service is this for?

Mastodon

### What are their domain names?

Looks like the Mastodon icon works for social.lol and mastodon.social, but `{mastodon}` doesn't work as a tag in the URL.

Might also be good to make it work for some other popular domains such as:
 - fosstodon.org
 - ???

Thanks!

**Comment by rosswintle** [comment]
OH! You need a space before the tag.

So `[Label](https://example.com/){icon}` doesn't work but `[Label](https://example.com/) {icon}` does!

Got it.

This looks like a little update needed to https://home.omg.lol/info/profile-items ?

**Comment by Mar0xy** [comment]
Btw this is already a duplicate of #189 aswell

**Comment by byemc** [comment]
@newbold consider maybe scraping the sites to see if they're a mastodon, then add that as an icon if it is? seems to be the most effective way of getting them all

**Comment by newbold** [comment]
@rosswintle Thanks for finding the bug with spaces before tags — I’ve just fixed that, so `foobar{tag}` and `foobar {tag}` will now work the same way. I've also updated the documentation with a better example.

I've also added `fosstodon.org` to the list of domains that will get a Mastodon icon by default. Glad to add others upon request, but given the rapidly expanding nature of the Fediverse and Mastodon instances in particular, I don’t think there’s any sane way that I could keep up with an accurate full list independently.

@Mar0xy and @byemc — I appreciate the sentiment behind your suggestions, but I think the technical overhead involved in trying to accomplish those things would be pretty high and I'm not sure it'd be worth the effort. Figuring out a way to reliably detect whether a URL points to a Mastodon instance seems like a tricky thing to do (especially in a way that would be future-proof), and it would also require me to start examining the content of every URL added to a profile — something that I don’t do today, and don’t really _want_ to do for a number of reasons. On the flip side, even if I had access to an accurate and always-updated list of Mastodon instances, it would likely have an enormous number of entries — and having to validate every URL used in a profile against that list would introduce inefficiencies that don't seem worthwhile, given that only a small percentage of the items in that list would ever be a match against an omg.lol profile entry.

I should probably more clearly spell out the philosophy and approach behind the profile icon process. The goal is to try to be helpful by applying a relevant icon to any given profile item _when it can be done easily_. I think that's what the goal _has_ to be, because if I said it was to apply an accurate icon to 100% of URLs entered then I'd be heading down a path that leads to endless toil and consistent disappointment. And even in the context of just trying to do this for Mastodon instances, it would be a never-ending effort that would always require extra effort to keep updated. In my view, it's better to just allow folks to tag whatever they wish, as well as to add an individual domain to the list of domains that get a Mastodon icon on an as-needed basis.
